python_modules/libraries/dagster-aws/dagster_aws/redshift/resources.py

```diff
@@ -1,23 +1,26 @@
 import abc
 from contextlib import contextmanager
+from typing import Optional, cast
 
 import psycopg2
 import psycopg2.extensions
 from dagster import (
-    Field,
-    IntSource,
-    StringSource,
     _check as check,
     resource,
 )
+from dagster._config.structured_config import (
+    ConfigurableResourceFactory,
+)
+from dagster._core.execution.context.init import InitResourceContext
+from pydantic import Field as PyField
 
 
 class RedshiftError(Exception):
     pass
 
 
-class _BaseRedshiftResource(abc.ABC):
-    def __init__(self, context):
+class BaseRedshiftClient(abc.ABC):
+    def __init__(self, context: InitResourceContext):
```

Comment:
So this predates you, but I don't think it makes sense for the Fake to inherit from a base where the constructor does real work like this. IMO we should heavily bias towards pure-interface-style ABCs.